### PR TITLE
Alisha/donation response

### PR DESCRIPTION
**Pull request by awu-hub**
# Pull Request

## Description

Added in drop downs for donation requests 

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

### Code Review Preparation
- [X] I have performed a self-review and reviewed the changed files
- [X] My code follows the project's style guidelines and passes linting
- [ ] I have added comments in hard-to-understand areas and updated documentation

### Testing (Ignore for the first PR)
- [ ] I have added relevant tests and they pass locally
- [X] I have tested the changes manually, including edge cases

### Code Quality & Security
- [X] I have formatted my code and removed debugging/unused code
- [X] No TypeScript errors or linting issues
- [X] No sensitive information committed (API keys, passwords, etc.)
- [ ] User inputs are validated and sanitized

### Dependencies & Deployment
- [ ] Dependencies are documented and package-lock.json is updated if needed
- [X] Changes work with Firebase emulator and environment variables are documentedRetry

## Screenshots (If it is a front end feature screenshot is required)

image

## Summary by CodeRabbit

* **New Features**
  * Add an "Acquisition" multiselect filter to donation request list views (Can Drop Off / Needs Pickup).
  * Table rows can show an inline Responded dropdown (Yes/No) to update responded status with immediate feedback.

* **UI Changes**
  * Details view now shows acquisition as a colored badge ("Can Drop Off" / "Needs Pickup").
  * Table row layouts toggle between responded control or quantity + review-progress display.

[Review Change Stack](https://app.coderabbit.ai/change-stack/Hack4Impact-UMD/journey-home/pull/81?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: components/inventory/DRDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                            checked={selected.includes(option) }
+                            onCheckedChange={checked => {
+                                if (!checked) return;
+                                setSelected([option]);
+
+                                setDonationRequest({
+                                    ...donationRequest, responded: option == "Yes",
+                                });
+                            }}
+                            onSelect={(event) => event.preventDefault()}
```
Comment (constrictingsnake) [inline components/inventory/DRDropdown.tsx]: currently the modal stays open after clicking something, when it should automatically close after selecting something

File: components/inventory/DRDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                        <DropdownMenuCheckboxItem
+                            key={option}
+                            className="text-xs cursor-pointer"
+                            checked={selected.includes(option) }
+                            onCheckedChange={checked => {
+                                if (!checked) return;
+                                setSelected([option]);
+
+                                setDonationRequest({
+                                    ...donationRequest, responded: option == "Yes",
```
Comment (constrictingsnake) [inline components/inventory/DRDropdown.tsx]: better to use === opposed to ==

File: components/donation-requests/DRDropdown.tsx
Comment (constrictingsnake) [inline components/donation-requests/DRDropdown.tsx:1]: Would make more sense for the DR dropdown component to be within donation-requests as opposed to inventory

**Review by constrictingsnake — CHANGES_REQUESTED** [review]

**Review by joelchem — APPROVED** [review]
